runtime/gateway.go

```diff
@@ -413,6 +424,25 @@ func (gateway *Gateway) setupLogger(config *StaticConfig) error {
 	return nil
 }
 
+func (gateway *Gateway) setupTracer(config *StaticConfig) error {
+	opts := []jaegerConfig.Option{
+		// TChannel logger implements jaeger logger interface
+		jaegerConfig.Logger(NewTChannelLogger(gateway.Logger)),
+		jaegerConfig.Metrics(NewJaegerMetricsFactory(gateway.MetricScope)),
+	}
+	// TODO: using default tracing configurations here, we may want to set it from static config
+	jc := jaegerConfig.Configuration{}
+	serviceName := config.MustGetString("serviceName")
+	closer, err := jc.InitGlobalTracer(serviceName, opts...)
```

Comment:
It's best to pass dependencies explicitly. It makes understanding the code easier and reduces complexity in writing tests.

Relying on singletons can create code that is near impossible to refactor without breaking everything :(